### PR TITLE
[WIP] Add panda robot

### PR DESCRIPTION
**Pull request by ykawamura96**
This PR is on top of previous PR: https://github.com/jsk-ros-pkg/jsk_robot/pull/1326 to include teleop_package (jsk_panda_teleop)

TODO:

- [x] add README.md to describe usage
- [x] add .rosinstall

### CODE REVIEW

File: jsk_panda_robot/jsk_panda_teleop/package.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>jsk_panda_teleop</name>
+  <version>0.0.0</version>
```
Comment (k-okada) [inline jsk_panda_robot/jsk_panda_teleop/package.xml:4]: package version must be as same as other packages in same repository, so please set to 1.1.0?

**Review by k-okada — CHANGES_REQUESTED** [review]
added wrong file?
```
A jsk_panda_robot/jsk_panda_accessories/tool_mounter/~$link_7.SLDPRT (0)
```

**Comment by ykawamura96** [comment]
> added wrong file?
> 
> ```
> A jsk_panda_robot/jsk_panda_accessories/tool_mounter/~$link_7.SLDPRT (0)
> ```

I removed the file. Thank you for the review.

File: jsk_panda_robot/jsk_panda.rosinstall [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Currently .rosintall is shared with user PC's one.
+
+
+# Wait until https://github.com/jsk-ros-pkg/jsk_robot/pull/1343 to be merged
+- git:
+    local-name: jsk-ros-pkg/jsk_robot
+    uri: https://github.com/ykawamura96/jsk_robot.git
+    version: add_panda_robot
+
+# Needed for using haptics devices
```
Comment (k-okada) [inline jsk_panda_robot/jsk_panda.rosinstall:11]: please add what is the difference between https://github.com/ykawamura96/Geomagic_Touch_ROS_Drivers and https://github.com/bharatm11/Geomagic_Touch_ROS_Drivers, and list of PRs that should be merged to use https://github.com/bharatm11/Geomagic_Touch_ROS_Drivers

File: jsk_panda_robot/jsk_panda_startup/euslisp/end-coords_publisher.l [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env roseus
```
Comment (k-okada) [inline jsk_panda_robot/jsk_panda_startup/euslisp/end-coords_publisher.l:1]: do we need this node?

File: jsk_panda_robot/jsk_panda_startup/launch/dual_panda.launch [hunk 4]
```diff
@@ -0,0 +1,121 @@
+<?xml version="1.0" ?>
+<launch>
+  <!-- uvc_cam -->
+  <!-- TODO: Use udev rule -->
+  <arg name="right_camera_dev" default="/dev/video0" />
+  <arg name="left_camera_dev" default="/dev/video2" />
```
Comment (k-okada) [inline jsk_panda_robot/jsk_panda_startup/launch/dual_panda.launch:6]: may be `/dev/video1` for defaults?

File: jsk_panda_robot/jsk_panda_startup/launch/panda_record.launch [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    <arg name="camera_namespace" value="$(arg camera_namespace)" />
+    <arg name="save_openni2" value="$(arg save_openni)" />
+    <arg name="save_openni" value="false" />
+    <arg name="save_robot_model" value="$(arg save_robot_model)" />
+    <arg name="save_all_image" value="$(arg save_all_image)" />
+    <arg name="save_multisense" value="$(arg save_multisense)" />
+    <arg if = "$(arg quiet)" name="other_options" value="-q"/>
+    <arg name="other_topics" value="$(arg other_topics)
+                                    /audio
+				                    /camera/depth/camera_info
```
Comment (k-okada) [inline jsk_panda_robot/jsk_panda_startup/launch/panda_record.launch:32]: mixing tab / spaces

File: jsk_panda_robot/jsk_panda_startup/scripts/panda_play.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```
Comment (k-okada) [inline jsk_panda_robot/jsk_panda_startup/scripts/panda_play.sh:1]: why we need this. for learning program?

File: jsk_panda_robot/jsk_panda_teleop/config/dual_panda_controllers.yaml [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+    arm_id: rarm
+    joint_names:
+      - rarm_joint1
+      - rarm_joint2
+      - rarm_joint3
+      - rarm_joint4
+      - rarm_joint5
+      - rarm_joint6
+      - rarm_joint7
+
```
Comment (k-okada) [inline jsk_panda_robot/jsk_panda_teleop/config/dual_panda_controllers.yaml:201]: add one more blank line?

File: jsk_panda_robot/jsk_panda_teleop/euslisp/move_to_initial.l [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  (send *robot* :angle-vector reset-pose)
+  (when (send *ri* :check-error)
+    (send *ri* :recover-error))
+  (send *ri* :angle-vector (send *robot* :angle-vector) 3000)
+  (send *ri* :wait-interpolation)
+  (send *ri* :stop-grasp :larm)
+  (send *ri* :stop-grasp :rarm))
+
+(main)
+(eixt)
```
Comment (k-okada) [inline jsk_panda_robot/jsk_panda_teleop/euslisp/move_to_initial.l:20]: typo?

File: jsk_panda_robot/jsk_panda_teleop/script/start_master_side.sh [hunk 9]
```diff
@@ -0,0 +1,24 @@
+
+function init_ros_ip () {
+    export ROS_MASTER_URI=http://133.11.216.227:11311
```
Comment (k-okada) [inline jsk_panda_robot/jsk_panda_teleop/script/start_master_side.sh:3]: we should write our IP settings on README.md and information on where should we fix, if we change IP address of devices.

File: jsk_panda_robot/panda_eus/euslisp/dual_panda-interface.l [hunk 10]
```diff
@@ -0,0 +1,380 @@
...
+  (send self :move-gripper arm 0.05 :effort 1.0 :wait wait)
+  )
+  (:get-table nil controller-table)
+  (:move-gripper
+  (arm pos &key (effort 25) (tm 500) (timeout 5000) (wait nil))  ;; TODO :arms is not implemented
+  (let* ((start-time 0.1) ;; TODO fixed start-time
+  (action (case arm
+  (:rarm r-gripper-action)
+  (:larm l-gripper-action)))
+  (joint-names (case arm
```
Comment (k-okada) [inline jsk_panda_robot/panda_eus/euslisp/dual_panda-interface.l:180]: indent?

File: jsk_panda_robot/jsk_panda.rosinstall (hunk 2 above)
Comment (pazeshun) [reply jsk_panda_robot/jsk_panda.rosinstall]: Done in #1543 and #1544

File: jsk_panda_robot/jsk_panda_startup/euslisp/end-coords_publisher.l (hunk 3 above)
Comment (pazeshun) [reply jsk_panda_robot/jsk_panda_startup/euslisp/end-coords_publisher.l:1]: Removed in #1543 and #1544

File: jsk_panda_robot/jsk_panda_startup/scripts/panda_play.sh (hunk 6 above)
Comment (pazeshun) [reply jsk_panda_robot/jsk_panda_startup/scripts/panda_play.sh:1]: Removed in #1543 and #1544

File: jsk_panda_robot/jsk_panda_teleop/script/start_master_side.sh (hunk 9 above)
Comment (pazeshun) [reply jsk_panda_robot/jsk_panda_teleop/script/start_master_side.sh:4]: Setting IP address manually is removed in #1543 and #1544

**Pull request closed without merging**